jhihyu/Knight_tour.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+import random
+
+# 8 steps
+next_step = [(2,1), (1,2), (2,-1), (1,-2), (-2,1), (-1,2), (-2,-1), (-1,-2)]
+
+# chessboard size
+board_size = int(input("Please enter the n of the n*n chessboard :"))
+
+
+class Knight_tour:
+
+    def __init__(self, size):
+        self.size = size
+        self.chessboard = np.zeros((self.size, self.size))
+        self.result = False
```

Comment:
It is better to move `self.result` as a return value of `travel` method, instead of a internal state of `Knight_tour` object.